### ISSUE TITLE
Concentrator executes multiple times when more than 1 GM user is logged in

### ISSUE THREAD

**Issue by BadIdeasBureau** [death-save/combat-utility-belt#460]
**Describe the bug**
On one particular Foundry instance (non-repro with identical settings & actors on a different server), Conentrator will consistently flag tokens for concentration two or three times, and has other intermittent issues where one copy of the concentration won't delete properly when this happens (even using macro, stays until server is reset).

**To Reproduce**
1. On the instance it affects, cast any concentration spell.
2. Non-reproducable on my own server

The only odd thing about the way that instance was set up was that settings were imported via Forien's Copy Environment, using the file here:  https://drive.google.com/file/d/1bxNAKeQ8mNZCIHbgy8EdP0lFsHNgh14p/view?usp=sharing , but the bug is non-repro when importing that into a new world on a different Foundry instance.  (the instance experiencing the bug is my GM's install, the settings dump is taken from my install and works fine on there).

The laptop the problem instance is running on was being very slow - though on the client side there was no really noticable issue with the server other than this.  I won't have a chance to reproduce until probably next weekend, but at that point might be able to check with the same /Data/ folder running on a different machine.

Probably sufficiently niche not to be an issue, but I'll try to keep digging if it persists.

**Expected behavior**
Concentration applies once, and goes away when dismissed.

**Screenshots**
See below.

**Environment Info (please complete the following information):**
 - Foundry Version: 0.7.9
 - Game System / Version: dnd5e 1.2.2
 - CUB Version 1.3.8
 - Browser: Affects clients on Chrome, Firefox, and in the electron app.
 - OS: Win10

**Additional context**
Occasionally, and not reliably reproducably, when deleting the duplicate conditions, one of them will remain on the token/actor.  When that happens, the only way to delete the condition is by refreshing the client or restarting the world (e.g. by saving module settings).  If the world is restarted, all goes back to normal.  If the client is refreshed but the world is not restarted, then the next concentration spell cast will show a chat message as if the character is already concentrating for only the first concentration condition applied, and the second concentration condition applied will apply normally.

Concentrator settings: (bug persists through deselecting settings and reapplying)
![image](https://user-images.githubusercontent.com/36359784/104132958-8a83b380-5378-11eb-9c20-350fd6db2549.png)

Example of multiple concentration from one cast - clicking to remove one removes both of them:
![image](https://user-images.githubusercontent.com/36359784/104132963-94a5b200-5378-11eb-8093-eb028f9e2561.png)

Example of single error following client restart when a condition is stuck on a character:
![image](https://user-images.githubusercontent.com/36359784/104133013-d59dc680-5378-11eb-9669-dd5d99bbff3c.png)

Console during spell cast - I couldn't manage to repro a concentration condition getting stuck with the console open (yellow warning errors are because DAE is disabled for this testing)
![image](https://user-images.githubusercontent.com/36359784/104133033-f534ef00-5378-11eb-8bfc-be59241d1b9c.png)

**Comment by eclarke12** [comment]
@BadIdeasBureau did you save the condition lab upon importing the settings? There's an important data massaging step that occurs on save that may not happen on load.

**Comment by xIHavocIx** [comment]
Came to second this issue. Started happening today and seems to be intermittent on which spells are causing it though I'm getting at least one that's doing it even after deleting and re-adding the spell item. 
![image](https://user-images.githubusercontent.com/22523890/106832448-52118580-6657-11eb-87ed-db191312019f.png)

Restarting the world seems to remove the glitched concentration that I cannot manually remove, but only until the spell is cast again.

**Comment by 8bitjunkie** [comment]
I have the same issue above consistently with the Friends (5e) spell. I don't know what's causing it, but I know how to fix it via the UI. Listing it here in case it helps.

For me, this issue occurs with spells that have the Concentration and a Target set to Self; all other Concentration spells work as expected.

How I fixed my issue:
- In 5e, the Self spell family defaults to a Target of none...so a dash "-" 
- If you look at its details on your sheet via Edit Item, you will see this (included pic)
- To fix my issue for this particular spell...I changed the dash "- " to a value of 1 via Edit Item and closed the editing panel.
- I then recast the Friends spell - now only 1 concentration condition pops and all is well - works as expected.
- When I go back into Edit Item on the Friends spell; I noticed the 1 was replaced back again with a dash "-"
- The Friends spell going forward works as intended with no more repeat conditions and I don't have to redo the steps I did above

Not sure why the above fixed my repeat conditions with Friends but thought I would share if it would help pinpoint the bug and/or conflict.
I have the DnDBeyond 5e Character Sheet mod in place; and yes I save my Condition Lab often and when prompted.

![image](https://user-images.githubusercontent.com/669735/107119384-6cf51d00-6844-11eb-8c4d-a2c6012172d1.png)

**Comment by Tappinator** [comment]
Hi,
I have this same issue but have determined it to happen only when there are more than 1 GM account logged into the server. Would it be possible to fix this?

**Comment by TheTercept** [comment]
Yep.  Have this same issue. When multiple DMs are logged in (Assistant DMs or straight Gamemaster) we get multiple concentrations set when spells are cast.

**Comment by Shuggaloaf** [comment]
I'd like to third the previous 2 commenters. I've also narrowed it down to the same thing - Multiple GMs / Assistant GMs cause duplicate conditions.

I see this is still listed as "unverified" even though @Tappinator and @TheTercept have also mentioned this as the cause of the issue back as far as Feb 28. This is very easy to reproduce. Just have more than 1 GM/Asst GM account logged in at the same time and cast a concentration spell. 

![image](https://user-images.githubusercontent.com/79543184/118331706-f0bf4500-b4d6-11eb-9f94-845ae5080b40.png)

**Comment by kuchrk24** [comment]
I just created an [issue ](https://github.com/death-save/combat-utility-belt/issues/599) about this without realizing there was some stuff here already. In my case I am the only GM. I don't have any assistants configured.
![image](https://user-images.githubusercontent.com/74018852/132619035-a531e693-e83e-46dc-af73-b0baea7f3891.png)

**Comment by eclarke12** [comment]
> I'd like to third the previous 2 commenters. I've also narrowed it down to the same thing - Multiple GMs / Assistant GMs cause duplicate conditions.
> 
> I see this is still listed as "unverified" even though @Tappinator and @TheTercept have also mentioned this as the cause of the issue back as far as Feb 28. This is very easy to reproduce. Just have more than 1 GM/Asst GM account logged in at the same time and cast a concentration spell.
> 
> ![image](https://user-images.githubusercontent.com/79543184/118331706-f0bf4500-b4d6-11eb-9f94-845ae5080b40.png)

`Unverified` tag simply means I haven't reproduced the issue yet. I'm not trying to diminish your experience or dismiss the issue!

**Comment by Shuggaloaf** [comment]
> `Unverified` tag simply means I haven't reproduced the issue yet. I'm not trying to diminish your experience or dismiss the issue!

Oh definitely didn't think that so no worries there! I see what I did now. When I first posted this I was new to get hub. I mis-understood the "assigning unverified tag" message and thought you had marked it that way. Now I know that is auto-assigned but never put 2 and 2 together. Sorry about that and thanks for the response. Either way I guess at least you know it's happening for more of us. Thanks!
